include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,405 @@
+{
+  template <class Extents>
+  class mapping {
+public:
+    using extents_type = Extents;
+    using index_type = typename extents_type::index_type;
+    using size_type = typename extents_type::size_type;
+    using rank_type = typename extents_type::rank_type;
+    using layout_type = layout_left_padded<padding_stride>;
+
+#ifndef MDSPAN_INTERNAL_TEST
+private:
+#endif // MDSPAN_INTERNAL_TEST
+
+    static_assert((padding_stride != 0) || (extents_type::static_extent(0) == 0), "padding stride can be 0 only if extents_type::static_extent(0) is 0");
+
+    static constexpr size_t __actual_padding_stride = detail::__get_actual_padding_stride<extents_type, padding_stride>();
+
+    using __inner_extents_type = typename detail::__inner_extents_left<extents_type, __actual_padding_stride>::__type;
+    using __unpadded_extent_type = typename detail::__unpadded_extent_type_impl<extents_type>::__type;
+    using __inner_mapping_type = layout_left::template mapping<__inner_extents_type>;
+
+    __inner_mapping_type __inner_mapping;
+    __unpadded_extent_type __unpadded_extent;
+
+public:
+#if !MDSPAN_HAS_CXX_20
+    MDSPAN_INLINE_FUNCTION_DEFAULTED
+    constexpr mapping()
+        : mapping(extents_type{})
+    {}
+#else
+    MDSPAN_INLINE_FUNCTION_DEFAULTED
+    constexpr mapping()
+      requires(__actual_padding_stride != dynamic_extent) = default;
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr mapping()
+      requires(__actual_padding_stride == dynamic_extent)
+        : mapping(extents_type{})
+    {}
+#endif
+
+    MDSPAN_INLINE_FUNCTION_DEFAULTED constexpr mapping(const mapping&) noexcept = default;
+    MDSPAN_INLINE_FUNCTION_DEFAULTED mapping& operator=(const mapping&) noexcept = default;
+
+    MDSPAN_INLINE_FUNCTION constexpr
+    mapping(const extents_type& __ext)
+      : __inner_mapping(detail::__inner_extents_left<extents_type, __actual_padding_stride>::template __construct<padding_stride>(__ext)),
+        __unpadded_extent(detail::__unpadded_extent_type_impl<extents_type>::__construct(__ext))
+    {}
+
+    template <class _Size>
+    // TODO: constraints
+    constexpr mapping(const extents_type &__ext, _Size __padding_value)
+        : __inner_mapping(detail::__inner_extents_left<extents_type, __actual_padding_stride>::template __construct<padding_stride>(__ext, static_cast<index_type>(__padding_value))),
+          __unpadded_extent(detail::__unpadded_extent_type_impl<extents_type>::__construct(__ext))
+    {
+      assert((padding_stride == dynamic_extent) || (padding_stride == static_cast<index_type>(__padding_value)));
+    }
+
+    template <class _OtherExtents>
+    constexpr MDSPAN_CONDITIONAL_EXPLICIT((!is_convertible_v<_OtherExtents, extents_type>))
+        mapping(const layout_left::mapping<_OtherExtents> &__other_mapping)
+        : __inner_mapping(detail::__inner_extents_left<extents_type, __actual_padding_stride>::template __construct_other<padding_stride>(__other_mapping.extents(), __other_mapping.stride(0))),
```

Comment:
doesn't this need to take `__other_mapping.stride(1)` I mean `stride(0)` is known as 1 for `layout_left` mappings.

Reply:
fixed